server/rosetta/converter_test.go

```diff
@@ -0,0 +1,355 @@
+	"github.com/stretchr/testify/suite"
+	crgerrs "github.com/tendermint/cosmos-rosetta-gateway/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+type ConverterTestSuite struct {
+	suite.Suite
+
+	c               rosetta.Converter
+	unsignedTxBytes []byte
+	unsignedTx      authsigning.Tx
+
+	util struct {
+		ir     codectypes.InterfaceRegistry
+		cdc    *codec.ProtoCodec
+		txConf client.TxConfig
+	}
+}
+
+func (s *ConverterTestSuite) SetupTest() {
+	// create an unsigned tx
+	const unsignedTxHex = "0a8e010a8b010a1c2f636f736d6f732e62616e6b2e763162657461312e4d736753656e64126b0a2d636f736d6f733134376b6c68377468356a6b6a793361616a736a3272717668747668396d666465333777713567122d636f736d6f73316d6e7670386c786b616679346c787777617175356561653764787630647a36687767797436331a0b0a057374616b651202313612600a4c0a460a1f2f636f736d6f732e63727970746f2e736563703235366b312e5075624b657912230a21034c92046950c876f4a5cb6c7797d6eeb9ef80d67ced4d45fb62b1e859240ba9ad12020a0012100a0a0a057374616b651201311090a10f1a00"
+	unsignedTxBytes, err := hex.DecodeString(unsignedTxHex)
+	s.Require().NoError(err)
+	s.unsignedTxBytes = unsignedTxBytes
+	// instantiate converter
+	cdc, ir := rosetta.MakeCodec()
+	txConfig := authtx.NewTxConfig(cdc, authtx.DefaultSignModes)
+	s.c = rosetta.NewConverter(cdc, ir, txConfig)
+	// add utils
+	s.util = struct {
+		ir     codectypes.InterfaceRegistry
+		cdc    *codec.ProtoCodec
+		txConf client.TxConfig
+	}{ir: ir, cdc: cdc, txConf: txConfig}
+	// add authsigning tx
+	sdkTx, err := txConfig.TxDecoder()(unsignedTxBytes)
+	s.Require().NoError(err)
+	builder, err := txConfig.WrapTxBuilder(sdkTx)
+	s.Require().NoError(err)
+
+	s.unsignedTx = builder.GetTx()
+}
+
+func (s *ConverterTestSuite) TestFromRosettaOpsToTxSuccess() {
+	addr1 := sdk.AccAddress("address1").String()
+	addr2 := sdk.AccAddress("address2").String()
+
+	msg1 := &bank.MsgSend{
+		FromAddress: addr1,
+		ToAddress:   addr2,
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin("test", 10)),
+	}
+
+	msg2 := &bank.MsgSend{
+		FromAddress: addr2,
+		ToAddress:   addr1,
+		Amount:      sdk.NewCoins(sdk.NewInt64Coin("utxo", 10)),
+	}
+
+	ops, err := s.c.ToRosetta().Ops("", msg1)
+	s.Require().NoError(err)
+
+	ops2, err := s.c.ToRosetta().Ops("", msg2)
+	s.Require().NoError(err)
+
+	ops = append(ops, ops2...)
+
+	tx, err := s.c.ToSDK().UnsignedTx(ops)
+	s.Require().NoError(err)
+
+	getMsgs := tx.GetMsgs()
+
+	s.Require().Equal(2, len(getMsgs))
+
+	s.Require().Equal(getMsgs[0], msg1)
+	s.Require().Equal(getMsgs[1], msg2)
+
+}
+
+func (s *ConverterTestSuite) TestFromRosettaOpsToTxErrors() {
+	s.Run("unrecognized op", func() {
+		op := &rosettatypes.Operation{
+			Type: "non-existent",
+		}
+
+		_, err := s.c.ToSDK().UnsignedTx([]*rosettatypes.Operation{op})
+
+		s.Require().ErrorIs(err, crgerrs.ErrBadArgument)
+	})
+
+	s.Run("codec type but not sdk.Msg", func() {
```

Comment:
this could be converted to table tests.

Reply:
@fdymylja can you address this in your follow up PR, please?